Ingress/controllers/nginx-third-party/README.md

```diff
@@ -0,0 +1,198 @@
+# Nginx Ingress Controller
+
+This is a nginx Ingress controller that uses [ThirdPartyResource](https://github.com/kubernetes/kubernetes/blob/master/docs/design/extending-api.md). See [Ingress controller documentation](../README.md) for details on how it works.
+
+## Disclaimer:
+
+This load balancer requires ThirdPartyResources enabled in apiserver.
+To do this specify the flag `--runtime-config=extensions/v1beta1/thirdpartyresources=true`
+
+
+## What it provides?
+
+- Ingress controller
+- nginx 1.9.x with [lua-nginx-module](https://github.com/openresty/lua-nginx-module)
+- SSL support
+- custom ssl_dhparam (optional). Just mount a secret with a file named `dhparam.pem`.
+- support for TCP services (flag `--tcp-services`)
+- custom nginx configuration using [ThirdPartyResource](https://github.com/kubernetes/kubernetes/blob/master/docs/design/extending-api.md)
+- custom error pages. Using the flag `--custom-error-service` is possible to use a custom compatible [404-server](https://github.com/kubernetes/contrib/tree/master/404-server) image [nginx-error-server](https://github.com/aledbf/contrib/tree/nginx-debug-server/Ingress/images/nginx-error-server) that provides an additional `/errors` route that returns custom content for a particular error code. **This is completely optional**
+
+
+## Requirements
+- default backend [404-server](https://github.com/kubernetes/contrib/tree/master/404-server) (or a compatible custom image)
+
+
+## SSL:
+
+Please follow [test.sh](https://github.com/bprashanth/Ingress/blob/master/examples/sni/nginx/test.sh) as a guide on how to generate secrets containing SSL certificates. The name of the secret can be different than the name of the certificate.
```

Comment:
suggest checking this in under a hack/ directory, it's simple enough as a script. What if i suddenly take off and delete my github ;)